### PR TITLE
Initial version of BLE central/Client

### PR DESCRIPTION
**Pull request by AdrianSoundy**
## Description

Initial implementation of Bluetooth Central/Client.

This covers the base implementation for : -

    Watching for Bluetooth Server devices
    Creating a BluetoothLEDevice object for the device
    Connecting & Discovering available services from device
    Discovering available characteristics in a service
    Discovering available descriptors in a characteristic
    Read / Write to an attribute ( Characteristic/Descriptor)
    Notify events for changes in attributes

See Readme.md for full details/examples code

Currently no support for pairing/bonding and authentication. This will be in next PR

## Motivation and Context
- Resolves nanoFramework/Home#981

## How Has This Been Tested?

Tested with sample code in separate PR. Will be available once main parts merges and nuget available.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: Tests/Client/Program.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        public static void Main()
+        {
+            Console.WriteLine();
+            Console.WriteLine("BlueTooth Test Client");
+
+            // Define used custom Uuid 
+            Guid serviceUuid = new("A7EEDF2C-DA87-4CB5-A9C5-5151C78B0066");
+            Guid readStaticCharUuid = new("A7EEDF2C-DA89-4CB5-A9C5-5151C78B0057");
+
+
```
Comment (josesimoes) [inline Tests/Client/Program.cs:47]:
```suggestion
```

File: nanoFramework.Device.Bluetooth.nuspec [hunk 2]
```diff
@@ -23,7 +23,8 @@ This package requires a target with nanoFramework.Device.Bluetooth v$nativeVersi
       <dependency id="nanoFramework.Runtime.Native" version="1.5.4" />
       <dependency id="nanoFramework.Runtime.Events" version="1.10.0" />
       <dependency id="nanoFramework.System.Text" version="1.2.3" />
-    </dependencies>
+	  <dependency id="nanoFramework.System.Collections" version="1.4.0" />
```
Comment (josesimoes) [inline nanoFramework.Device.Bluetooth.nuspec:26]: This identation looks a bit off...

File: nanoFramework.Device.Bluetooth/BluetoothEventListener.cs [hunk 3]
```diff
@@ -5,52 +5,104 @@
...
 using nanoFramework.Device.Bluetooth.GenericAttributeProfile;
+using nanoFramework.Device.Bluetooth.Advertisement;
 
 namespace nanoFramework.Device.Bluetooth
 {
     internal class BluetoothEventListener : IEventProcessor, IEventListener
     {
         // Map of Bluetooth Characteristic numbers to GattLocalCharacteristic objects.
-        private static readonly ArrayList _characteristicMap = new ArrayList();
+        private static readonly ArrayList _characteristicMap = new();
```
Comment (josesimoes) [inline nanoFramework.Device.Bluetooth/BluetoothEventListener.cs:19]: PascalCase in static field names please.

**Review by josesimoes — COMMENTED** [review]
WOW again! Lots of code and nice sample usage on the readme. 👍🏻

File: nanoFramework.Device.Bluetooth.nuspec (hunk 2 above)
Comment (AdrianSoundy) [reply nanoFramework.Device.Bluetooth.nuspec]: fixed

**Comment by AdrianSoundy** [comment]
Issues resolved so merging now :-)

**Pull request merged**